doc/mgmt/config_replace_rollback_HLD.md

```diff
@@ -0,0 +1,416 @@
+	- Delete backup config file.
+	- Unmark trial stage.
+	- Close config session.  
+	  
+
+# 4.3 Config Session mode configuration commit confirm command
+![Config_session_commit_confirm](config_replace_images/Config_commit_confirm.png "Figure 4: Config Session mode commit confirm")  
+
+ __Figure 4: Config Session mode commit confirm__  
+Extend config session commit rpc for:  
+- Delete backup config file.  
+- Unmark trial stage.	
+- Close config session.  
+	
+
+# 5 Error Handling
+- Primary Encryption cannot be replaced using config replace key config-key password-encrypt and
+no key config-key password-encrypt commands trigger re-encryption of keys in backends and much more. 
+If Primary Encryption related commands are Created/Modified/Deleted through config replace,
+Error will be thrown
+	
+- User management commands cannot be replaced using config replace username and no username commands 
+directly configure local users on linux. If User management commands are Created/Modified/Deleted through config replace,
+Error will be thrown
+	
+- Break out of physical ports cannot be supported. If Break out related commands are Created/Modified/Deleted 
+through config replace, Error will be thrown 
+
+# 6 Serviceability and Debug
+
+
+# 7 Scalability
+
+
+# 8 Platform
+
+
+# 9 Security and Threat Model
+
+
+## 9.1 Certificates, secrets, keys, and passwords
+
+
+# 10 Limitations
+
+# 11 Unit Test
+
+# 12 Customer Use Cases  
+The following uses cases are verified as part of POC.
+
+## 12.1 TACACS+
+- Configure multiple tacacs server from exec mode using config replace.
+- Unconfigure a tacacs server from exec mode using config replace.
+- Configure multiple tacacs server from config session mode using replace and commit.
+- Unconfigure a tacacs server from config session mode using replace and commit.
+- Verify the functionality of tacacs after each configure/unconfigure scenario.
+
+## 12.2 ACL
+- Configure access-list with multiple permit/deny statements from exec mode using config replace.
+- Unconfigure permit/deny statements from exec mode using config replace.
+- Configure access-list with multiple permit/deny statements from config session mode using replace and commit.
+- Unconfigure permit/deny statements from config session mode using replace and commit.
+- Verify the functionality of access-lists after each configure/unconfigure scenario.
+
+## 12.3 Portchannel
+- Add interface members to the portchannel from exec mode using config replace.
+- Delete one or some of the members from the portchannel from exec mode using config replace.
+- Add interface members to the portchannel from config session mode using replace and commit.
+- Delete one or some of the members from the portchannel from config session mode using replace and commit.
+- Verify the members are added and deleted successfully to/from the portchannel and traffic disruption should be negligible due to portchannel re-hashing.
+
+## 12.4 BGP
```

Comment:
Is BGP neighbor add/delete the only BGP config update required by the customers?

Reply:
At the time of writing this HLD, the PoC limits the validation scope to the specific asks from Google and our sales team. More scaled, complex tests will be covered in the dev-test.